masu/celery/tasks.py

```diff
@@ -16,96 +16,22 @@
 #
 """Celery task definitions."""
 
-# pylint: disable=unused-argument, fixme, unused-import
-# FIXME: temporary module-wide disable until tasks are fully implemented.
-
+import datetime
+import logging
 from unittest import mock
 
+from celery.task import periodic_task
 from celery.utils.log import get_task_logger
 
-from masu import create_app
-from masu.celery import create_celery
-from masu.processor.exceptions import MasuProcessingError, MasuProviderError
+from masu.processor.tasks.download import get_report_files
+from masu.processor.tasks.process import process_report_file
+from masu.processor.orchestrator import Orchestrator
 
-MASU = create_app()
-MASU.app_context().push()
-CELERY = create_celery(MASU)
```

Comment:
@blentz If I did not misunderstand, I think this is the circular-y parts that you referenced? If so, we don't need to instantiate an app or a celery instance here, we can define a task, and the `autodiscover_tasks` call in the app factory should handle associating the tasks.

Reply:
I am also curious where the circular dependency was in the design.  It was one thing I was very aware of and made every effort to avoid that when I was working on the orchestrator and integrating the downloader/processor.  If I missed something please let me know.

Reply:
@adberglund - That's correct. This file was the source of circular imports.

Reply:
@dccurtis The circular dependency ends up being because of how Celery uses decorators to mark tasks.

To get access to the decorators, you need a celery instance.
To get a celery instance, you can import the `CELERY` instance right there.
But, importing CELERY is a problem when, say, the `api.download` view has already imported the `processor.tasks` module that you're trying to mark with the `@CELERY.task` decorators.